src/Resources/public/js/pricing/config/item.js

```diff
@@ -362,24 +362,34 @@ pimcore.bundle.EcommerceFramework.pricing.config.item = Class.create({
                 var item = conditions[i].items.getAt(c);
 
                 try {
+                    let value = null;
                     // workaround for pimcore.object.tags.objects
                     if(item.reference)
                     {
-                        condition[ item.reference.getName() ] = item.reference.getValue();
+                        value = item.reference.getValue();
+                        condition[ item.reference.getName() ] = value;
                     }
                     else if(item.form)
                     {
-                        condition[ item.name ] = item.getForm().getFieldValues();
+                        value = item.form.getFieldValues();
+                        condition[ item.name ] = value;
                     }
                     else if(item.xtype === 'datefield')
                     {
-                        condition[ item.name ] = item.getSubmitValue();
+                        value = item.getSubmitValue();
+                        condition[ item.name ] = value;
                     }
                     else
                     {
-                        condition[ item.getName() ] = item.getValue();
+                        value = item.getValue();
+                        condition[ item.getName() ] = value;
 
                     }
+
+                    if (item.mandatory && (!value)){
+                        Ext.MessageBox.alert(t("error"), t("mandatory_field_empty"));
```

Comment:
I would also include field-name in the error message as the asterisk is anyway missing.

Reply:
![image](https://github.com/pimcore/ecommerce-framework-bundle/assets/6014195/d32a56c2-a2fd-4b41-b23f-6dc2ccc557a3)

Reply:
the good is it avoids 
![image](https://github.com/pimcore/ecommerce-framework-bundle/assets/6014195/1a4339d6-c252-4650-8820-a245925beba3)

to reproduce, i just add `mandatory: true,` to https://github.com/pimcore/ecommerce-framework-bundle/blob/36eb2c132f35132cea9c031d34a714edb1555c17/src/Resources/public/js/pricing/config/item.js#L986 and use the `Cart Amount`  condition as the screenshot